light/client.go

```diff
@@ -258,6 +269,24 @@ func NewClientFromTrustedStore(
 	return c, nil
 }
 
+// isBlacklisted checks whether provider is black listed
+// NOTE: requires a providerMutex lock
```

Comment:
Yea @udpatil discussed this with Yiming too, this occurs in other parts of the code as well - reference `removeWitnesses` . For now, since both fns are only internal not public, its not a large concern. 
There isn't a straightforward way to guarantee caller has the lock - unless we just move all this logics directly into `Add Provider` . For now will leave as is and can discuss a refactor later